core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -619,6 +663,44 @@ object Chunk
     override def map[O2](f: O => O2): Chunk[O2] = indexedSeq(s.map(f))
   }
 
+  /** Creates a chunk from a mutable `java.util.List`. */
+  def javaList[O](javaList: ju.List[O]): Chunk[O] =
+    javaList match {
+      case chunkAsJavaList: ChunkAsJavaList[O] =>
+        chunkAsJavaList.chunk
+
+      case randomAccess: ju.RandomAccess =>
+        new JavaListChunk(randomAccess)
+
+      case _ =>
+        val size = javaList.size
+        val arr = new Array[Object](size).asInstanceOf[Array[O with Object]]
+        javaList.toArray(arr)
+        new ArraySlice(arr, 0, size)(ClassTag.Object.asInstanceOf[ClassTag[O with Object]])
```

Comment:
If it is ok to make a `Chunk` that is backed by a mutable collection (as we discussed above), then is it really necessary to make a copy here? I mean, since `Chunk` does not guarantee efficient random access:
https://github.com/typelevel/fs2/blob/5dbe235e5a4e763251f94c3880a75ba20915edbb/core/shared/src/main/scala/fs2/Chunk.scala#L1029

So is it that necessary to enforce it here then?

Reply:
@satorg uhm, that is a good point, not sure then.
@armanbilge @mpilquist should we just always use `JavaListAsChunk`?

Reply:
I would keep a copy in this case. Chunk.Queue is the only case we've relaxed O(1) indexed access. I'd like to keep it the only case.